plugins/analytics/public/domainiq.py

```diff
@@ -0,0 +1,126 @@
+import json
+import logging
+import requests
+from core.observables import Ip, Hostname, Email, Text
+from core.analytics import OneShotAnalytics
+from core.errors import GenericYetiError, ObservableValidationError
+
+mapped_types = {
+    "Ip": "ip_report",
+    "Hostname": "domain_report",
+    "Email": "email_report",
+}
+
+class DomainIQApi(object):
+    """Base class for querying the DomainIQ API."""
+
+    @staticmethod
+    def fetch(observable, apikey):
+        params = {
+            "key": apikey,
+            "output_mode": "json",
+        }
+
+        if isinstance(observable, Ip):
+            params["service"] = "ip_report"
+            params["ip"] = observable.value
+        elif isinstance(observable, Hostname):
+            params["service"] = "domain_report"
+            params["domain"] = observable.value
+        elif isinstance(observable, Email):
+            params["service"] = "email_report"
+            params["email"] = observable.value
+
+        try:
+            r = requests.get("https://www.domainiq.com/api", params=params)
+            if r.ok:
+                return r.json()
+            else:
+                raise GenericYetiError("{} - {}".format(r.status_code, r.content))
+        except Exception as e:
+            logging.erro(e)
+            raise GenericYetiError("{} - {}".format(r.status_code, r.content))
+
+
+class DomainIQ(DomainIQApi, OneShotAnalytics):
+    default_values = {
+        'name': 'DomainIQ',
+        'description': 'Perform a DomainIQ query.',
+    }
+
+    settings = {
+            "domainiq_apikey": {
+                "name": "DomainIQ apikey",
+                "description": "apikey for domainiq."
+            }
+    }
+
+    ACTS_ON = ['Hostname', 'Ip', 'Email']
+
+    @staticmethod
+    def process_response(observable, json_result):
+        links = set()
+
+        if isinstance(observable, Ip):
```

Comment:
Please refactor in methods for each case you're dealing with (Ip, Hostname, Email)